gm4_mountaineering/data/gm4_mountaineering/functions/main.mcfunction

```diff
@@ -0,0 +1,13 @@
+# Run ski-related functions
+execute as @a[gamemode=!spectator] at @s run function gm4_mountaineering:ski_effects/skis_equipped
+
+# Adds jump boost effects when holding poles
+execute as @a[gamemode=!spectator,predicate=gm4_mountaineering:holding_poles] at @s run function gm4_mountaineering:ski_effects/poles_jump_boost
+
+# Adds vertical climbing abilities when wearing crampons and not holding anything
+execute as @a[gamemode=!spectator,predicate=gm4_mountaineering:wearing_crampons] at @s run function gm4_mountaineering:climbing_effects/crampons_equipped
```

Comment:
This function runs every time for players wearing Crampons, but not everything in the called function needs to run every time. It's better to call a separate "initiate Crampons" function first that adds a tag to the player and does things that only need to happen once like adding the speed increase Attribute.
Anything that does need to be on a clock can then just run on players with that tag.